src/OpenLoco/Vehicles/VehicleBogie.cpp

```diff
@@ -1,16 +1,145 @@
+        }
+        else if (!(vehicleUpdate_var_1136114 & (1 << 2)))
+        {
+            return true;
+        }
+
+        sub_4AA464();
+        explodeComponent();
+        var_0C |= Flags0C::unk_5;
+        var_5A &= ~(1u << 31);
+        var_5A >>= 3;
+        var_5A |= (1u << 31);
+
+        Vehicle train(head);
+        bool end = false;
+        for (auto& car : train.cars)
+        {
+            for (auto& carComponent : car)
+            {
+                if (carComponent.front == this || carComponent.back == this)
+                {
+                    carComponent.body->explodeComponent();
+                    carComponent.body->var_5A &= ~(1u << 31);
+                    carComponent.body->var_5A >>= 3;
+                    carComponent.body->var_5A |= (1u << 31);
+                    end = true;
+                    break;
+                }
+            }
+            if (end)
+            {
+                break;
+            }
+        }
+
+        auto* collideEntity = EntityManager::get<EntityBase>(vehicleUpdate_collisionCarComponent);
+        auto* collideCarComponent = collideEntity->asVehicle();
+        if (collideCarComponent != nullptr)
+        {
+            Vehicle collideTrain(collideCarComponent->getHead());
+            if (collideTrain.head->status != Status::crashed)
+            {
+                collideCarComponent->sub_4AA464();
+            }
+
+            end = false;
+            for (auto& car : train.cars)
+            {
+                for (auto& carComponent : car)
+                {
+                    if (carComponent.front == collideCarComponent)
+                    {
+                        carComponent.front->explodeComponent();
+                        carComponent.front->var_5A &= ~(1u << 31);
+                        carComponent.front->var_5A >>= 3;
+                        carComponent.front->var_5A |= (1u << 31);
+                    }
+                    if (carComponent.back == collideCarComponent)
+                    {
+                        carComponent.back->explodeComponent();
+                        carComponent.back->var_5A &= ~(1u << 31);
+                        carComponent.back->var_5A >>= 3;
+                        carComponent.back->var_5A |= (1u << 31);
+                    }
+                    if (carComponent.front == collideCarComponent || carComponent.back == collideCarComponent || carComponent.body == collideCarComponent)
+                    {
+                        carComponent.body->explodeComponent();
+                        carComponent.body->var_5A &= ~(1u << 31);
+                        carComponent.body->var_5A >>= 3;
+                        carComponent.body->var_5A |= (1u << 31);
+                        end = true;
+                        break;
+                    }
+                }
+                if (end)
+                {
+                    break;
+                }
+            }
+        }
+        return false;
```

Comment:
all of this seems to basically be something like `HandleVehicleCollision`, can we put it in a separate method?